sqe-http-api/Services/UserService.cs

```diff
@@ -104,16 +122,110 @@ public UserDTO GetCurrentUser()
+to activate your new account is: $Token.<br>
+<br>
+Best wishes,<br>
+The Scripta Qumranica Electronica team</body></html>";
+            const string emailSubject = "Activation of your Scripta Qumranica Electronica account";
+            var name = !string.IsNullOrEmpty(newUserData.forename) || !string.IsNullOrEmpty(newUserData.surname)
+                ? (newUserData.forename + " " + newUserData.surname).Trim() 
+                : newUserData.userName;
+            await _emailSender.SendEmailAsync(
+                newUserData.email,
+                emailSubject,
+                emailBody.Replace("$User", name)
+                    .Replace("$Token", createdUser.Token)
+                );
+            return UserModelToDTO(createdUser);
+        }
+
+        /// <summary>
+        /// Confirms the registration of a new user via a secret authentication token, which
+        /// was sent to the user's email address.
+        /// </summary>
+        /// <param name="token">Secret authentication token for user's new account</param>
+        /// <returns></returns>
+        public async Task ConfirmUserRegistrationAsync(string token)
+        {
+            await _userRepository.ConfirmAccountCreationAsync(token);
+        }
+        
+        public async Task RequestResetLostPasswordAsync(string email)
+        {
+            var userInfo = await _userRepository.RequestResetForgottenPasswordAsync(email);
+            if (userInfo == null) // Silently return on error
+                return;
+            
+            // Email the user
+            // TODO: Add link to web endpoint when we know what that is. Can token be in URL query?
+            const string emailBody = @"
+<html><body>Dear $User,<br>
+<br>
+Sorry to hear that you have lost your password for Scripta Qumranica Electronica.  You may reset your password with the token: $Token.<br>
+<br>
+Best wishes,<br>
+The Scripta Qumranica Electronica team</body></html>";
+            const string emailSubject = "Lost password for your Scripta Qumranica Electronica account";
+            await _emailSender.SendEmailAsync(
+                email,
+                emailSubject,
+                emailBody.Replace("$User", userInfo.UserName)
+                    .Replace("$Token", userInfo.Token)
+            );
+            return;
+        }
+
+        /// <summary>
+        /// Change password from old password to new password for the user's account.
+        /// </summary>
+        /// <param name="user">User object with all information for the user requesting a password change</param>
+        /// <param name="oldPassword">The old password for the user's account</param>
+        /// <param name="newPassword">The new password for the user's account</param>
+        /// <returns></returns>
+        public async Task ChangePasswordAsync(UserInfo user, string oldPassword, string newPassword)
+        {
+            await _userRepository.ChangePasswordAsync(user, oldPassword, newPassword);
```

Comment:
Do we want to email someone when their password has been changed? A lot of services do that.

Reply:
That is a nice idea.  I have added that.

Done.